webapp/src/app/components/widgets/event-widget/event-registry/show-modal-choice.ts

```diff
@@ -0,0 +1,78 @@
+import { Label } from '../../../../models/events';
+import { EntityAttributes } from '../../../../services/phaser/entities/cc-entity';
+import { AbstractEvent, EventType } from './abstract-event';
+
+export interface ShowModalChoiceData extends EventType {
+    [key: number]: AbstractEvent<any>[];
+
+    text: Label;
+    options: {
+        label: Label;
+    }[];
+}
+
+export class ShowModalChoice extends AbstractEvent<ShowModalChoiceData> {
+    private attributes: EntityAttributes = {
+        text: {
+            type: 'LangLabel',
+            description: 'Text of modal dialog',
+			hs: true
+        },
+        options: {
+            type: 'ModalChoiceOptions',
+            description: 'All the options of the modal dialog',
+			C2: true
+        }
+    };
+
+    getAttributes(): EntityAttributes {
+        return this.attributes;    
+    }
+
+    update() {
+        this.children = [];
+		this.info = this.combineStrings(
+			this.getTypeString('#7ea3ff'),
```

Comment:
Is this color good? I feel like it should be a similar color but I can understand making it different.

Also I found out the source of the `[object Object]`, it was because of how `getPropString()` does not properly handle a falsy `value` being passed in. It was easy to fix though.
![image](https://github.com/CCDirectLink/crosscode-map-editor/assets/49847844/3e0da7a8-9533-48d0-b3bd-3ca037fd1957)